src/Microsoft.VisualStudio.Services.Agent/TraceSourceWrapper.cs

```diff
@@ -0,0 +1,119 @@
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Newtonsoft.Json;
+using System;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.VisualStudio.Services.Agent
+{
+    public sealed class TraceSourceWrapper : IDisposable
```

Comment:
I don't like TraceSourceWrapper referenced everywhere.  This is effectively our tracing class.  Can we call it AgentTrace?  Tracing?